FilePersistence/src/version_control/Commit.cpp

```diff
@@ -103,4 +103,88 @@ bool Commit::getFileContent(QString fileName, const char*& content, int& content
 		return false;
 }
 
+QStringList Commit::regexSearchNodeData(QString id) const
```

Comment:
Also this function should accept directly an `NodeIdType` and convert it to `QString` locally.